common/lockmanager.go

```diff
@@ -0,0 +1,121 @@
+package common
+
+import (
+    "fmt"
+    "sync"
+    "time"
+
+    "github.com/edulinq/autograder/config"
+    "github.com/edulinq/autograder/log"
+)
+
+type lockData struct {
+    timestamp time.Time;
+    mutex sync.RWMutex;
+    isLocked int;
```

Comment:
Change this to something that indicates its non-binary nature.
Maybe something like "lockCount"?